qiime2/sdk/plugin_manager.py

```diff
@@ -100,44 +127,52 @@ def _integrate_plugin(self, plugin):
 
             self.formats[name] = record
         self.type_formats.extend(plugin.type_formats)
+        self.types[plugin.name] = plugin.types
```

Comment:
I'm not sure if this structure has any real advantage over `self.plugins[<name>].types`, so I would drop it.